src/zap.ts

```diff
@@ -0,0 +1,55 @@
+import { ChainId, isEthereum, isFantom } from "./chain";
+import { EthAddress, ZeroAddress } from "./helpers";
+import { Token, VaultMetadataOverrides } from "./types";
+
+export type ZapInWith = keyof Token["supported"];
+
+type ZapInDetails = { isZapInSupported: boolean; zapInWith?: ZapInWith };
+
+type ZapInArgs = {
+  chainId: ChainId;
+  token?: Partial<Token>;
+  vaultMetadata: VaultMetadataOverrides | null;
+};
+
+export function getZapInDetails({ chainId, token, vaultMetadata }: ZapInArgs): ZapInDetails {
+  if (!token?.supported || !vaultMetadata?.zapInWith) {
+    return { isZapInSupported: false };
+  }
+
+  if (isEthereum(chainId)) {
+    if (token.address === EthAddress) {
+      return { isZapInSupported: Boolean(token.supported.zapper), zapInWith: "zapperZapIn" };
+    }
```

Comment:
Since not all vaults accept zaps, it doent matter if the token is ETH, since ETH also needs the zapping contract to be supported. We can just remove this piece of code and manage ETH zaps as any other. (The only thing different with ETH is that it does not require approve)

Reply:
Removed with https://github.com/yearn/yearn-sdk/pull/280/commits/fe35a937e3de5a5e84b9061ebda2028213652a45